src/async-wrap.cc

```diff
@@ -429,13 +433,19 @@ static void SetupHooks(const FunctionCallbackInfo<Value>& args) {
 
 static void EnablePromiseHook(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
-  env->AddPromiseHook(PromiseHook, nullptr);
+  env->AddPromiseHook(PromiseHook, static_cast<void*>(env));
 }
 
 
 static void DisablePromiseHook(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
-  env->RemovePromiseHook(PromiseHook, nullptr);
+
+  // Delay the call to `RemovePromiseHook` because we might currently be
```

Comment:
Since we are going with the temp fix, I guess this is okay. But I really want to see a better solution. I'm not really convinced there aren't some obscure bug with the right `.enable()` and `.disable()` calls.